### PR TITLE
Per-style delaying

### PR DESCRIPTION
**Pull request by owahab**
Adding ability to delay a specific style by adding :delayed => true to a specific style.
Breaks current delay-all-by-default behavior.

### CODE REVIEW

**Comment by owahab** [comment]
Updated code to restore the default behavior since it makes more sense.

File: lib/delayed/paperclip.rb [hunk 1]
```diff
@@ -14,18 +14,28 @@ module Delayed
 
         define_method "halt_processing_for_#{name}" do
           return unless self.send("#{name}_changed?")
-
-          false # halts processing
+          if self.send(:id).nil?
```
Comment (jstorimer) [inline lib/delayed/paperclip.rb]: Why not use this?

``` ruby
if new_record? 
```

**Comment by jstorimer** [comment]
I'm interested in merging this, but two things:

1) I'm not going to merge a version bump, can you remove that commit?
2) I would really like to have tests for this. I realize that the testing env was a bit of a pain to get set up, but I've made some changes tonight that should make that process easier.

**Comment by owahab** [comment]
1) I pumped the version number because I introduced a new major feature. Will remove it.
2) I would love to do so as well. I will try your new changes and will keep you posted.

**Comment by jstorimer** [comment]
I agree that when this gets released it will need a version bump. But I'd rather make that call after looking over all the changes, and having a chance to try things out in the master branch without changing the version.

**Comment by owahab** [comment]
Roger that. Wilco.

File: lib/delayed/paperclip.rb (hunk 1 above)
Comment (owahab) [inline lib/delayed/paperclip.rb]: Ignorance perhaps :)

**Comment by owahab** [comment]
Still unable to add tests due to issue #27

**Comment by Bertg** [comment]
Sorry for the inconvenience, but could you split these 2 changes and rebase them on the current master?

There have been quite some changes that should be taken into account...

**Pull request closed without merging**